modules/internal/DefaultRectangular.chpl

```diff
@@ -809,6 +1013,241 @@ module DefaultRectangular {
+      ref where tag == iterKind.standalone && !defRectSimpleDData {
+      if debugDefaultDist {
+        chpl_debug_writeln("*** In array standalone code");
+      }
+      for i in dom.these(tag, tasksPerLocale,
+                         ignoreRunning, minIndicesPerTask) {
+        yield dsiAccess(i);
+      }
+    }
+
+    iter these(param tag: iterKind,
+               tasksPerLocale = dataParTasksPerLocale,
+               ignoreRunning = dataParIgnoreRunningTasks,
+               minIndicesPerTask = dataParMinGranularity)
+      where tag == iterKind.leader && !defRectSimpleDData {
+      // This was adapted from the DefaultRectangularDom leader.
+      const dptpl = if tasksPerLocale==0 then here.maxTaskPar
+                    else tasksPerLocale;
+      if debugDataParMultiDData {
+        chpl_debug_writeln("### mdNumChunks = ", mdNumChunks, "\n",
+                           "### ignoreRunning = ", ignoreRunning, "\n",
+                           "### minIndicesPerTask = ", minIndicesPerTask, "\n",
+                           "### mdParDim = ", mdParDim, "\n",
+                           "### dom.dsiDims = ", dom.dsiDims());
+      }
+      coforall chunk in 0..#mdNumChunks { // make sure coforall on can trigger
+        local on here.getChild(chunk) { // eventually, on dataChunk(chunk).locale
+          if debugDataParMultiDData {
+            if chunk != chpl_getSubloc() then
+              writeln("*** ERROR: multiDD:  ON WRONG SUBLOC (should be ",
+                      chunk, ", on ", chpl_getSubloc(), ") ***");
+          }
+          // Divide the locale's tasks approximately evenly
+          // among the chunks.
+          const numSublocTasks = (if chunk < dptpl % mdNumChunks
+                                  then dptpl / mdNumChunks + 1
+                                  else dptpl / mdNumChunks);
+          var locBlock: rank*range(idxType);
+          for param i in 1..rank do
+            locBlock(i) = 0:idxType..#(dom.dsiDim(i).length);
+          var followMe = locBlock;
+          const (lo,hi) = _computeBlock(locBlock(mdParDim).length,
+                                        mdNumChunks, chunk,
+                                        locBlock(mdParDim).high,
+                                        locBlock(mdParDim).low,
+                                        locBlock(mdParDim).low);
+          followMe(mdParDim) = lo..hi;
+          const (numChunks2, parDim2) = _computeChunkStuff(numSublocTasks,
+                                                           ignoreRunning,
+                                                           minIndicesPerTask,
+                                                           followMe);
+          if debugDataParMultiDData then
+            chpl_debug_writeln("### multiDD: chunk ", chunk,
+                               ", followMe ", followMe,
+                               ", numChunks2 ", numChunks2);
+          coforall chunk2 in 0..#numChunks2 do local on here {
```

Comment:
I think the 'local' is just saying that the on-stmt won't do anything across the network.  I don't believe it precludes referencing other sublocales, though.